dcore/state.py

```diff
@@ -75,32 +81,58 @@ def dump(self):
             self.output.dumplist = self.fieldlist
 
         funcs = self.xn.split()
+        field_dict = {name: func for (name, func) in zip(self.fieldlist, funcs)}
         to_dump = []
-
-        for name, f in zip(self.fieldlist, funcs):
+        for name, f in field_dict.iteritems():
             if name in self.output.dumplist:
                 to_dump.append(f)
             f.rename(name=name)
 
-        for field, meanfield in zip(funcs, self.output.meanfields):
+        steady_state_dump_err = self.output.steady_state_dump_err
+        for field in self.fieldlist:
+            steady_state_dump_err.setdefault(field, False)
```

Comment:
I think I can't use defaultdict here because I need specific keys in the dictionary - is that correct?